backend/controllers/searchWithFilter.test.js

```diff
@@ -0,0 +1,396 @@
+        allowDownload.mockReturnValue(true)
+
+        const req = {
+          ...standardReq,
+          query: {},
+        }
+
+        const render = jest.fn()
+        await search(req, { render })
+        expect(render).toBeCalledWith(
+          'searchWithFilter.njk',
+          expect.objectContaining({
+            downloadUrl: '/search-with-filter-dps-users/download?user=&status=ALL&roleCode=&groupCode=&activeCaseload=',
+          }),
+        )
+      })
+      it('should not call renderer with download url when download is not allowed', async () => {
+        allowDownload.mockReturnValue(false)
+
+        const req = {
+          ...standardReq,
+          query: {},
+        }
+
+        const render = jest.fn()
+        await search(req, { render })
+        expect(render).toBeCalledWith(
+          'searchWithFilter.njk',
+          expect.not.objectContaining({
+            downloadUrl: '/search-with-filter-dps-users/download?user=&status=ALL&roleCode=&groupCode=&activeCaseload=',
+          }),
+        )
+      })
+
+      it('should set current filter with single query parameters', async () => {
+        const req = {
+          ...standardReq,
+          query: {
+            user: 'Andy',
+            status: 'INACTIVE',
+            groupCode: 'MDI',
+            roleCode: 'ACCESS_ROLE_ADMIN',
+            restrictToActiveGroup: 'false',
+          },
+        }
+
+        const render = jest.fn()
+        await search(req, { render, locals: { user: { maintainAccessAdmin: true } } })
+        expect(render).toBeCalledWith(
+          'searchWithFilter.njk',
+          expect.objectContaining({
+            currentFilter: {
+              groupCode: ['MDI'],
+              roleCode: ['ACCESS_ROLE_ADMIN'],
+              status: 'INACTIVE',
+              user: 'Andy',
+              restrictToActiveGroup: false,
+            },
+            downloadUrl:
+              '/search-with-filter-dps-users/download?user=Andy&status=INACTIVE&roleCode=ACCESS_ROLE_ADMIN&groupCode=MDI&activeCaseload=',
+          }),
+        )
+      })
+
+      it('should set list current filters to undefined when no values', async () => {
+        const req = {
+          ...standardReq,
+          query: { user: 'Andy', status: 'INACTIVE', groupCode: '', roleCode: '', restrictToActiveGroup: '' },
+        }
+
+        const render = jest.fn()
+        await search(req, { render, locals: { user: { maintainAccessAdmin: true } } })
+        expect(render).toBeCalledWith(
+          'searchWithFilter.njk',
+          expect.objectContaining({
+            currentFilter: {
+              groupCode: undefined,
+              roleCode: undefined,
+              status: 'INACTIVE',
+              user: 'Andy',
+              restrictToActiveGroup: true,
+            },
+            downloadUrl:
+              '/search-with-filter-dps-users/download?user=Andy&status=INACTIVE&roleCode=&groupCode=&activeCaseload=',
```

Comment:
This might be considered a nice to have - but would it be better if undefined filters are not included in the query parameters?

Reply:
Agreed, but kept with how it functions now, they are "sent" as empty strings - I wasn't sure of the implications of changing that (though 98% it would be fine). But did a like for like